grow/cache/podcache.py

```diff
@@ -105,9 +108,14 @@ def _write_json(self, path, obj):
         output = json.dumps(
             obj, cls=json_encoder.GrowJSONEncoder,
             sort_keys=True, indent=2, separators=(',', ': '))
-        temp_path = '{}.tmp'.format(path)
-        self._pod.write_file(temp_path, output)
-        self._pod.move_file_to(temp_path, path)
+        _, temp_path = tempfile.mkstemp()
+        with open(temp_path, 'w') as fp:
```

Comment:
Somewhat of a micro-optimization but it's better to open the file descriptor since it's already available to the process.

```
fd, path = tempfile.mkstemp()
with os.fdopen(fd, 'w') as fp:
  fp.write(output)
```

(you also won't need to make sure the dir exists with this since it's guaranteed to be writeable)

Reply:
Updated; but flagging the "makedirs" is for the "real path", not the "temp path". In testing it looks like the real path can be a folder that doesn't exist when this is invoked (since it might go into the `.grow` folder).